R/parseOutput.R

```diff
@@ -1704,6 +1704,65 @@ extractTech10 <- function(outfiletext, filename) {
   
   tech10List <- list()
   
+  bivarFit <- getSection("^\\s*BIVARIATE MODEL FIT INFORMATION\\s*$", outfiletext)
+  
+  if (is.null(bivarFit)) return(tech10List) 
+  
```

Comment:
Interesting approach, thanks for taking the time to write it. My personal view is that it makes what the code is doing harder to understand, for a probably fairly small benefit in performance (at least in real world use).

However, happy to defer to the package maintainers about which version is preferred.

Reply:
I was just trying to show how to parse this section using a vectorized approach. R doesn't work well with for-loops and repeated assignment to temporary variables. This example is already more than 4x as fast than the looped version.

Reply:
We do get some complaints regarding performance so if it can be vectorized, I'd vote for that.

To help with us taking this onboard, can someone post either (1) an Mplus output script so we can test OR (2) some Mplus code that runs a relevant model?

Thanks!

Josh

Reply:
The 7.3 example from the Mplus website has TECH10 output:

https://www.statmodel.com/usersguide/chap7/ex7.3.html

Performance could potentially be an issue, as these sections can become quite long if there are a number of indicator variables with several categories each.